generator/src/googleapis/codegen/template_objects.py

```diff
@@ -128,6 +128,35 @@ def __init__(self, def_dict, api, parent=None, wire_name=None,
       self.SetTemplateValue('description',
                             self.ValidateAndSanitizeComment(self.StripHTML(d)))
 
+  def ComputeNonDuplicatedName(self, class_name):
+    parent_path = self.parentPath()
+    occurrences = 0
+
+    def increment_name(s, n):
+      """
+      Increments the number (if any) of the duplicated class name
+      :param s: class_name
+      :param n: occurrences
+      :return: new class_name with incremented occurrence (e.g. Test1 to Test2
+      or Test to Test1)
+      """
+      return s[0: - len(str(n - 1)) if n > 1 else len(s)] + str(n), n
+
+    # increase occurrences as they occur in the path
+    # from root class to current one
+    for parent_class_name in parent_path:
+      if class_name == parent_class_name:
+        class_name, occurrences = increment_name(class_name, occurrences + 1)
+
+    # increase occurrences as they occur in sibling resources
+    if self.parent is not None:
```

Comment:
If there are multiple siblings, is this going to give all of the siblings the same name?

Reply:
Yes, the class name "increments" for each of the siblings

Reply:
Used new logic: append `Resource` if the sibling is a resource or `Request` if it's a method

Reply:
If the duplication can only because of `Resource` or `Method`, can we only append `Request` and leave `Resource` as it is? Also I remember `Method` are actually a collection of requests, so maybe `Requests` makes more sense?

Reply:
There was a mistake in the generation from the last time we checked the code. Actually, it happens that `Method` refers to a single method. For example, a `User` resource may have the methods `DeleteUser` and `CreateUser`. There are no nested methods or resources inside a method.

I agree with leaving Resources intact.